packages/extension/src/serviceWorker/index.ts

```diff
@@ -17,790 +17,373 @@
  */
-import { fetchDictionary } from '../utils/fetchCookieDictionary';
-import {
-  ALLOWED_NUMBER_OF_TABS,
-  DEVTOOLS_CLOSE,
-  DEVTOOLS_OPEN,
-  DEVTOOLS_SET_JAVASCSCRIPT_COOKIE,
-  INITIAL_SYNC,
-  POPUP_CLOSE,
-  POPUP_OPEN,
-  SERVICE_WORKER_RELOAD_MESSAGE,
-  SERVICE_WORKER_TABS_RELOAD_COMMAND,
-  SET_TAB_TO_READ,
-  TABID_STORAGE,
-} from '../constants';
-import SynchnorousCookieStore from '../store/synchnorousCookieStore';
-import { getTab } from '../utils/getTab';
-import parseHeaders from '../utils/parseHeaders';
-import resetCookieBadgeText from '../store/utils/resetCookieBadgeText';
-import getQueryParams from '../utils/getQueryParams';
-import reloadCurrentTab from '../utils/reloadCurrentTab';
-
-let cookieDB: CookieDatabase | null = null;
-let syncCookieStore: SynchnorousCookieStore | undefined;
-
-const requestIdToCDPURLMapping: {
-  [tabId: string]: {
-    [requestId: string]: string;
-  };
-} = {};
-
-let tabMode: 'single' | 'unlimited' = 'single';
-let tabToRead = '';
-let globalIsUsingCDP = false;
+import syncCookieStore from '../store/synchnorousCookieStore';
+import createCookieFromAuditsIssue from '../utils/createCookieFromAuditsIssue';
+import attachCDP from './attachCDP';
+import './chromeListeners';
 
 const ALLOWED_EVENTS = [
   'Network.responseReceived',
   'Network.requestWillBeSentExtraInfo',
   'Network.responseReceivedExtraInfo',
   'Audits.issueAdded',
+  'Network.requestWillBeSent',
+  'Page.frameAttached',
+  'Page.frameNavigated',
+  'Target.attachedToTarget',
 ];
 
-/**
- * Fires when the browser receives a response from a web server.
- * @see https://developer.chrome.com/docs/extensions/reference/api/webRequest
- */
-chrome.webRequest.onResponseStarted.addListener(
-  ({ tabId, url, responseHeaders, frameId, requestId }) => {
-    (async () => {
-      const tab = await getTab(tabId);
-      let tabUrl = syncCookieStore?.getTabUrl(tabId);
-
-      // Sometimes, a site may send out requests while it is still in the preloading state. Any cookie set from these requests are classified as third-party cookies.
-      // For example nikkei.com. The cookie domain may be nikkei.com however the tab URL would be xyz.com so it becomes third-party
-      if (tab && tab.pendingUrl) {
-        tabUrl = tab.pendingUrl;
-      }
-
-      if (!cookieDB) {
-        cookieDB = await fetchDictionary();
-      }
-
-      const cookies = await parseHeaders(
-        globalIsUsingCDP,
-        'response',
-        tabToRead,
-        tabMode,
-        tabId,
-        url,
-        cookieDB,
-        tabUrl,
-        frameId,
-        requestId,
-        responseHeaders
-      );
-
-      if (!cookies || (cookies && cookies?.length === 0)) {
-        return;
-      }
-
-      // Adds the cookies from the request headers to the cookies object.
-      syncCookieStore?.update(tabId, cookies);
-    })();
-  },
-  { urls: ['*://*/*'] },
-  ['extraHeaders', 'responseHeaders']
-);
+let targets = await chrome.debugger.getTargets();
```

Comment:
What's the purpose of getting the target here when it is being overridden at line number 60?

Reply:
Fixed in latest commit. I guess this was left when service worker was refactored.